source/performance/F0.Minesweeper.Logic.Benchmarks/LocationShuffler/LocationShufflerBenchmarks.cs

```diff
@@ -0,0 +1,69 @@
+using System.Diagnostics;
+using F0.Minesweeper.Logic.Abstractions;
+using F0.Minesweeper.Logic.LocationShuffler;
+
+namespace F0.Minesweeper.Logic.Benchmarks.LocationShuffler
+{
+	public class LocationShufflerBenchmarks
+	{
+		private readonly ILocationShuffler globallyUniqueIdentifier = new GuidLocationShuffler();
+		private readonly ILocationShuffler randomOrder = new RandomOrderLocationShuffler();
+		private readonly ILocationShuffler fisherYates = new FisherYatesLocationShuffler();
+
+		private IEnumerable<Location> minefield = null!;
+
+		[ParamsSource(nameof(Parameters))]
+		public Param Parameter { get; set; } = null!;
+
+		[GlobalSetup]
+		public void GlobalSetup()
+		{
+			uint size = Parameter.Width * Parameter.Height;
+			List<Location> locations = new((int)size);
+
+			for (uint width = 0; width < Parameter.Width; width++)
+			{
+				for (uint height = 0; height < Parameter.Height; height++)
+				{
+					locations.Add(new Location(width, height));
+				}
+			}
+
+			Debug.Assert(locations.Count == size);
+			minefield = locations;
+		}
+
+		[Benchmark]
+		public IReadOnlyCollection<Location> GloballyUniqueIdentifier()
+			=> globallyUniqueIdentifier.ShuffleAndTake(minefield, (int)Parameter.MineCount);
+
+		[Benchmark]
+		public IReadOnlyCollection<Location> RandomOrder()
+			=> randomOrder.ShuffleAndTake(minefield, (int)Parameter.MineCount);
+
+		[Benchmark]
+		public IReadOnlyCollection<Location> FisherYates()
+			=> fisherYates.ShuffleAndTake(minefield, (int)Parameter.MineCount);
+
+		public static IEnumerable<Param> Parameters()
+		{
+			MinefieldOptions[] collection = new[]
```

Comment:
If we want to we could drop the `new[]` too

Reply:
We have some more usages of `new[]` (_30_ occurrences in this branch) - Do we want to remove/keep it everywhere?

Reply:
Yes - we want to remove them where possible - and so it was done.